### PR TITLE
Add README and LICENSE

### PR DESCRIPTION
**Pull request by brianhlin**
I migrated [HowToWriteTests](https://twiki.opensciencegrid.org/bin/view/SoftwareTeam/HowToWriteTests) into the `README.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+The whole system uses the standard Python `unittest` framework to run. Note that all tests have to be compatible with Python 2.4; when reading the docs for `unittest`, keep note of when a feature was introduced.
+
+### Test Sequence
+
+During a test run, the test modules are run in sequence as follows:
+
+| File                 | When                                          | Purpose                                                               |
+|:---------------------|:----------------------------------------------|:----------------------------------------------------------------------|
+| `special_user.py`    | Tests not suppressed, or explicitly requested | Add user (if asked)&lt;br&gt;Check user&lt;br&gt;Set up mapfile       |
```
Comment (matyasselmeci) [inline README.md]: The <br> tags got translated into HTML entities

Comment (brianhlin) [inline README.md]: Good catch, fixed (as well as another place I missed some more HTML)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+2.  Get rid of the old tests:
+    ```
+    # For RHEL 6, CentOS 6, and SL6 or OSG 3 older than 3.1.15
+    [root@client ~]$ rm -rf /usr/lib/python2.6/site-packages/osgtest
+    # For RHEL 7, CentOS 7, and SL7
+    [root@client ~]$ rm -rf /usr/lib/python2.7/site-packages/osgtest
+    ```
+3.  `cd` into your clone of the `osg-test` repo and copy over your tests to your VM: 
+    ```
+    # For RHEL 6, CentOS 6, and SL6 VMs<span class="twiki-macro ENDCOLOR"></span> <span class="twiki-macro UCL_PROMPT"></span>
```
Comment (matyasselmeci) [inline README.md]: A `<span>` thing got in here.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+Each test module must import the `osgunittest` library, plus whichever of the `osg-test` libraries are needed (conventionally with shortened aliases):
+
+```python
+import osgunittest
+
+import osgtest.library.core as core
+import osgtest.library.files as files
+```
+
+Then, a single test class is defined, deriving from =osgunittest.OSGTestCase=; the individual tests are sequentially numbered functions within the class:
```
Comment (matyasselmeci) [inline README.md]: Replae `=` around `osgunittest.OSGTestCase` with backticks. (If anyone knows how to quote a backtick in markdown, tell me!)

**Comment by matyasselmeci** [comment]
There are some content changes I'd like to suggest -- should those go into a separate PR, though?

File: README.md (hunk 3 above)
Comment (brianhlin) [inline README.md]: One of these what you're looking for?

```
\`foo`
```
\`foo`

```
> \`foo`
```

> \`foo`

**Comment by brianhlin** [comment]
I think content changes should go into its own github issue and/or a PR. If you're satisfied with it otherwise, go ahead and approve the PR!

**Comment by matyasselmeci** [comment]
LGTM. GitHub is not giving me the option to approve the review, so I'll just merge it.

**Comment by djw8605** [comment]
I don't see the license actually applied anywhere, just the LICENSE file.  I think at the bottom of the readme, you should have the actual license and copyright wording.

**Comment by brianhlin** [comment]
I don't think that's strictly necessary, GitHub automagically slurps `LICENSE` or `LICENSE.md` files:

https://help.github.com/articles/adding-a-license-to-a-repository/

[HTCondor-CE](https://github.com/opensciencegrid/htcondor-ce) does it this way

**Comment by bbockelm** [comment]
It is indeed best practice to add it to individual files - but I think that adding a license to the repository itself is a fine (sufficient?) first step.

**Comment by djw8605** [comment]
Sure, but there is no copyright notice anywhere.  Isn't that part of the point of a license file?

**Comment by bbockelm** [comment]
Copyright and licensing are a bit orthogonal (unless the author signs over the copyright as part of the code submission).

It is indeed traditional to have a copyright statement in the LICENSE file, but that made a lot more sense in the case where there was a clear single owner of the copyright (i.e., "Board of Regent, University of California").  Makes less sense for distributed projects

**Pull request merged**